src/spritekit.cs

```diff
@@ -2372,5 +2558,372 @@ interface SKReferenceNode {
+
+		[Export ("centerOfTileAtColumn:row:")]
+		CGPoint GetCenterOfTile (nuint column, nuint row);
+	}
+
+	[iOS (10,0)][Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface SKTileSet : NSCopying, NSCoding
+	{
+		[Static]
+		[Export ("tileSetWithTileGroups:")]
+		SKTileSet Create (SKTileGroup[] tileGroups);
+
+		[Static]
+		[Export ("tileSetWithTileGroups:tileSetType:")]
+		SKTileSet Create (SKTileGroup[] tileGroups, SKTileSetType tileSetType);
+
+		[Export ("initWithTileGroups:")]
+		IntPtr Constructor (SKTileGroup[] tileGroups);
+
+		[Export ("initWithTileGroups:tileSetType:")]
+		IntPtr Constructor (SKTileGroup[] tileGroups, SKTileSetType tileSetType);
+
+		[Static]
+		[Export ("tileSetNamed:")]
+		[return: NullAllowed]
+		SKTileSet FromName (string name);
+
+		[Static]
+		[Export ("tileSetFromURL:")]
+		[return: NullAllowed]
+		SKTileSet FromUrl (NSUrl url);
+
+		[Export ("tileGroups", ArgumentSemantic.Copy)]
+		SKTileGroup[] TileGroups { get; set; }
+
+		[NullAllowed, Export ("name")]
+		string Name { get; set; }
+
+		[Export ("type", ArgumentSemantic.Assign)]
+		SKTileSetType Type { get; set; }
+
+		[NullAllowed, Export ("defaultTileGroup", ArgumentSemantic.Assign)]
+		SKTileGroup DefaultTileGroup { get; set; }
+
+		[Export ("defaultTileSize", ArgumentSemantic.Assign)]
+		CGSize DefaultTileSize { get; set; }
+	}
+
+	[iOS (10,0)][Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface SKTileGroup : NSCopying, NSCoding
+	{
+		[Static]
+		[Export ("tileGroupWithTileDefinition:")]
+		SKTileGroup Create (SKTileDefinition tileDefinition);
+
+		[Static]
+		[Export ("tileGroupWithRules:")]
+		SKTileGroup Create (SKTileGroupRule[] rules);
+
+		[Static]
+		[Export ("emptyTileGroup")]
+		SKTileGroup CreateEmpty ();
+
+		[Export ("initWithTileDefinition:")]
+		IntPtr Constructor (SKTileDefinition tileDefinition);
+
+		[Export ("initWithRules:")]
+		IntPtr Constructor (SKTileGroupRule[] rules);
+
+		[Export ("rules", ArgumentSemantic.Copy)]
+		SKTileGroupRule[] Rules { get; set; }
+
+		[NullAllowed, Export ("name")]
+		string Name { get; set; }
+	}
+
+	[iOS (10,0)][Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface SKTileGroupRule : NSCopying, NSCoding
+	{
+		[Static]
+		[Export ("tileGroupRuleWithAdjacency:tileDefinitions:")]
+		SKTileGroupRule Create (SKTileAdjacencyMask adjacency, SKTileDefinition[] tileDefinitions);
+
+		[Export ("initWithAdjacency:tileDefinitions:")]
+		IntPtr Constructor (SKTileAdjacencyMask adjacency, SKTileDefinition[] tileDefinitions);
+
+		[Export ("adjacency", ArgumentSemantic.Assign)]
+		SKTileAdjacencyMask Adjacency { get; set; }
+
+		[Export ("tileDefinitions", ArgumentSemantic.Copy)]
+		SKTileDefinition[] TileDefinitions { get; set; }
+
+		[NullAllowed, Export ("name")]
+		string Name { get; set; }
+	}
+
+	[iOS (10,0)][Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface SKWarpGeometry : NSCopying, NSCoding {}
+
+	interface ISKWarpable {}
+
+	[iOS (10,0)][Mac (10,12)]
+	[Protocol, Model]
```

Comment:
@spouliot by "concrete model" you are commenting the fact that I added `interface ISKWarpable {}` which, I agree, will likely not be used in the bindings or the use of `[Model]` which, to me, is required because of the base type.